Mathlib/LinearAlgebra/DirectSum/Finsupp.lean

```diff
@@ -12,20 +12,274 @@ import Mathlib.LinearAlgebra.DirectSum.TensorProduct
 /-!
 # Results on finitely supported functions.
 
-The tensor product of `ι →₀ M` and `κ →₀ N` is linearly equivalent to `(ι × κ) →₀ (M ⊗ N)`.
+* `TensorProduct.finsuppLeft`, the tensor product of `ι →₀ M` and `N`
+  is linearly equivalent to `ι →₀ M ⊗[R] N`
+
+* `TensorProduct.finsuppScalarLeft`, the tensor product of `ι →₀ R` and `N`
+  is linearly equivalent to `ι →₀ N`
+
+* `TensorProduct.finsuppRight`, the tensor product of `M` and `ι →₀ N`
+  is linearly equivalent to `ι →₀ M ⊗[R] N`
+
+* `TensorProduct.finsuppScalarRight`, the tensor product of `M` and `ι →₀ R`
+  is linearly equivalent to `ι →₀ N`
+
+* `TensorProduct.finsuppLeft'`, if `M` is an `S`-module,
+  then the tensor product of `ι →₀ M` and `N` is `S`-linearly equivalent
+  to `ι →₀ M ⊗[R] N`
+
+* `finsuppTensorFinsupp`, the tensor product of `ι →₀ M` and `κ →₀ N`
+  is linearly equivalent to `(ι × κ) →₀ (M ⊗ N)`.
+
+## Case of MvPolynomial
+
+These functions apply to `MvPolynomial`, one can define
+```
+noncomputable def MvPolynomial.rTensor' :
+    MvPolynomial σ S ⊗[R] N ≃ₗ[S] (σ →₀ ℕ) →₀ (S ⊗[R] N) :=
+  TensorProduct.finsuppLeft'
+
+noncomputable def MvPolynomial.rTensor :
+    MvPolynomial σ R ⊗[R] N ≃ₗ[R] (σ →₀ ℕ) →₀ N :=
+  TensorProduct.finsuppScalarLeft
+ ```
+
+However, to be actually usable, these definitions need lemmas to be given in companion PR.
+
+## Case of `Polynomial`
+
+`Polynomial` is a structure containing a `Finsupp`, so these functions
+can't be applied directly to `Polynomial`.
+
+Some linear equivs need to be added to mathlib for that.
+This belongs to a companion PR.
+
+## TODO
+
+* generalize to `MonoidAlgebra`, `AlgHom `
+
+* reprove `TensorProduct.finsuppLeft'` using existing heterobasic version of `TensorProduct.congr`
 -/
 
 
 noncomputable section
 
-open DirectSum Set LinearMap Submodule TensorProduct
+open DirectSum TensorProduct
+
+open Set LinearMap Submodule
+
+section TensorProduct
+
+variable {R : Type*} [CommSemiring R]
+  {M : Type*} [AddCommMonoid M] [Module R M]
+  {N : Type*} [AddCommMonoid N] [Module R N]
+
+namespace TensorProduct
+
+variable {ι : Type*} [DecidableEq ι]
+
+/-- The tensor product of `ι →₀ M` and `N` is linearly equivalent to `ι →₀ M ⊗[R] N` -/
+noncomputable def finsuppLeft :
+    (ι →₀ M) ⊗[R] N ≃ₗ[R] ι →₀ (M ⊗[R] N) :=
+  (congr (finsuppLEquivDirectSum R M ι) (LinearEquiv.refl R N)).trans
+    ((directSumLeft R (fun _ : ι => M) N).trans
+      (finsuppLEquivDirectSum R (M ⊗[R] N) ι).symm)
```

Comment:
I'd replace `.trans` by `≪≫ₗ`. Do you prefer `trans`?

Reply:
yes, it's much faster to type to me…

Reply:
Anyway, I changed trans to that. It spares some parentheses…